network/config.go

```diff
@@ -61,12 +62,25 @@ func DefaultConfig() *Config {
 	}
 }
 
+func validateAddresses(address []string) error {
+	for _, addr := range address {
+		_, err := multiaddr.NewMultiaddr(addr)
+		if err != nil {
+			fmt.Println("Failed to parse address: ", err)
+			return errors.Errorf(errors.ErrInvalidAddress, "invalid relay and listen address")
+		}
+	}
+	return nil
+}
+
 // SanityCheck performs basic checks on the configuration.
 func (conf *Config) SanityCheck() error {
 	if conf.EnableRelay {
 		if len(conf.RelayAddrs) == 0 {
 			return errors.Errorf(errors.ErrInvalidConfig, "at least one relay address should be defined")
 		}
 	}
+	validateAddresses(conf.Listens)
```

Comment:
```
if err := validateAddresses(conf.Listens); err != nil {
    return err
}
```

Reply:
should I add this for `conf.RelayAddrs` too?
```go
if err := validateAddresses(conf.Listens); err != nil {
    return err
}
if err := validateAddresses(conf.RelayAddrs); err != nil {
    return err
}
```
like this